ledger/src/blockstore.rs

```diff
@@ -803,8 +805,9 @@ impl Blockstore {
                 ErasureMetaStatus::DataFull => {
                     (set_index..set_index + erasure_meta.config.num_coding() as u64).for_each(
                         |i| {
-                            // Remove saved coding shreds. We don't need these for future recovery.
-                            if prev_inserted_codes.remove(&(slot, i)).is_some() {
+                            // Instead of removing the coding block, we keep it to allow serving
+                            // coding shreds over repair.
+                            if prev_inserted_codes.contains_key(&(slot, i)) {
```

Comment:
need to remove `index.coding_mut().set_present(i, false);` line below to keep index consistent